### ISSUE TITLE
Assume non-public add-on when the API returns a 404

### ISSUE THREAD

**Issue by willdurand** [mozilla/addons-frontend#12539]
In https://github.com/mozilla/addons-frontend/issues/12478, we added some logic to make a non-public add-on when the API call to fetch an add-on returns a 401 or a 403. We should do the same for a 404.

mat said:

> An example on dev is `{34f676d6-32ef-4833-adff-f84e992bb7c5}` which was listed before getting deleted. (Note: if you're logging in with the right permissions you'd be able to load it anyway. Make sure to be logged out when testing)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/ADDFRNT-261)

**Comment by eviljeff** [comment]
I'm not saying this is the correct or incorrect choice but from the matrix conversation I didn't glean a (Product type) decision had been made on how it should work?  @diox I don't see where you said that.

**Comment by diox** [comment]
We discussed that on Matrix earlier.

I don't think a product decision is required, this looks like a fairly basic use-case to me: You should be able to report an add-on that you have installed but has since been deleted (regardless of the reason), since we may want to block it. As far as the user is concerned this isn't really different from a disabled/pure unlisted add-on.

**Comment by ioanarusiczki** [comment]
@willdurand I don't seem to get the results I've mentioned this morning https://github.com/mozilla/addons-frontend/issues/12478#issuecomment-1809808156 but since I hit the IP throttle I'll have to repeat testing for unauthenticated reports or with a regular user. 

I tried with invisible , disabled as a developer or deleted from dev hub and now after page refresh I get the add-on form like this
https://addons-dev.allizom.org/en-US/firefox/feedback/addon/only-firefox/
-> no name, author or rating stars but the slug is available not the guid. 
Meanwhile, an admin can still see the name and the author (without the ratings on the right) 

It's a bit confusing, I'll leave it for tomorrow.

**Comment by ioanarusiczki** [comment]
@willdurand 

Repeated some testing and this is what I've noticed:

- for listed add-ons if I open the report as a regular user (or unauthenticated) I'll see the form, name, author, ratings. If meanwhile the public addon becomes unlisted (developer's actions or a reviewer's action takes place) I refresh the feedback form and I'll see https://addons-dev.allizom.org/en-US/firefox/addon/webextension-with-image-3/ or https://addons-dev.allizom.org/en-US/firefox/feedback/addon/api-compatibility-checks4/. Trying to send a report from such a page won't work , the slug needs to be replaced with guid. 

- for unlisted versions reports can be sent only with guid. 

I think this is expected. Clearly when I tested yesterday I replaced the slug with the add-on guid and I continued to do testing on the same add-on, failed to notice what is happening at a page refresh for the listed.
